sky/backends/backend_utils.py

```diff
@@ -458,6 +439,7 @@ def add_cluster(
         auth_config: Dict[str, str],
         ports: List[int],
         docker_user: Optional[str] = None,
+        ssh_user: Optional[str] = None,
```

Comment:
It should be fine to remove the `ssh_user` from this function arguments, but it might be hard to remove it from the `handle` as we have to keep that information for the cluster without changing the cluster_yaml. : )